src/main/scala/sigmastate/utils/ByteWriter.scala

```diff
@@ -12,8 +12,39 @@ trait ByteWriter {
   def put(x: Byte): ByteWriter
   def putBoolean(x: Boolean): ByteWriter
   def putShort(x: Short): ByteWriter
+
```

Comment:
@ergomorphic I'd appreciate your feedback for the writer/reader API changes below. I've made them with this one commit. Basically, use VLQ with ZigZag in `putInt`, `putLong` and add `putUInt`, `putULong` with pure VLQ for positive values encoding. See scaladoc in `ByteArrayWriter` methods for details.